solrmarc_mixin/src/de/unituebingen/ub/ubtools/solrmarcMixin/TuelibMixin.java

```diff
@@ -996,4 +983,508 @@ public String getRecordingDate(final Record record) {
+        }
+
+        return formats;
+    }
+
+    /**
+     * Determine Format(s) but do away with the electronic tag
+     *
+     * @param record
+     *            the record
+     * @return mediatype of the record
+     */
+
+    public Set<String> getFormat(final Record record) {
+        Set<String> formats = getFormatIncludingElectronic(record);
+
+        // Since we now have an additional facet mediatype we remove the
+        // electronic label
+        formats.remove("Electronic");
+
+        return formats;
+    }
+
+    /**
+     * Determine Mediatype For facets we need to differentiate between
+     * electronic and non-electronic resources
+     *
+     * @param record
+     *            the record
+     * @return mediatype of the record
+     */
+
+    public Set<String> getMediatype(final Record record) {
+        final Set<String> mediatype = new HashSet<>();
+        final Set<String> formats = getFormatIncludingElectronic(record);
+        final String electronicRessource = "Electronic";
+        final String nonElectronicRessource = "Non-Electronic";
+
+        if (formats.contains(electronicRessource)) {
+            mediatype.add(electronicRessource);
+        } else {
+            mediatype.add(nonElectronicRessource);
+        }
+
+        return mediatype;
+    }
+
+    /**
+     * Determine the publication date for "date ascending/descending" sorting in
+     * accordance with the rules stated in issue 227
+     *
+     * @param record
+     *            MARC record
+     * @return the publication date to be used for
+     */
+
+    public String getPublicationSortDate(final Record record) {
+
+        final Set<String> format = getFormatIncludingElectronic(record);
+
+        // Case 1 [Article or Review]
+        // Match also the case of publication date transgressing one year
+        // (Format YYYY/YY for older and Format YYYY/YYYY) for
+        // newer entries
+        if (format.contains("Article") || format.contains("Review")) {
+            final DataField _936Field = (DataField) record.getVariableField("936");
+            if (_936Field != null) {
+                final Subfield jSubfield = _936Field.getSubfield('j');
+                if (jSubfield != null) {
+                    final String yearOrYearRange = jSubfield.getData();
+                    return yearOrYearRange.length() > 4 ? yearOrYearRange.substring(0, 4) : yearOrYearRange;
+                } else {
+                    System.err.println("getPublicationSortDate [No matching subfield 'j' in field 936]: " + record.getControlNumber());
+                }
+            } else {
+
+                System.err.println("getPublicationSortDate [No matching 936 field:] " + record.getControlNumber());
+            }
+
+        // Case 2: Otherwise get from 260c 
+        } else {
+            final Set<String> dates = new LinkedHashSet<>();
+            final List<VariableField> list260 = record.getVariableFields("260");
```

Comment:
I think we usually name variables like this _260Fields etc.  Also below, vf and df are not really good names.  Using the type of a variable as its name is not good practice.  Please use what the variable stores, its function etc.